### PR TITLE
Fix LAMA - don't return `truth`

### PR DESCRIPTION
**Pull request by PGijsbers**
When processing the results, if `truth` is returned it is processed based on `is_encoded` - regardless of if only the predictions are encoded or also truth ([here](https://github.com/openml/automlbenchmark/blob/c6eb3784ddde723fc4c7a3230886b1a67d142cb3/amlb/results.py#L290-L298)).

I'm not entirely sure what the original reason was, but it seems to me that there is no good reason for allowing the frameworks return `truth` (or even pass labels of the test data in the first place)? I'd be curious to know why it's a thing (and consequently see if perhaps we should rework that in the future).

### CODE REVIEW

**Review by sebhrusen — APPROVED** [review]
OK, I didn't get that lightsautoml was returning encoded predictions even when the input target is not encoded.
I just hope that they're encoded the same way as we usually do it so that we can decode them correctly: based on `iris` it looks ok, I hope there's no edge case.

**Comment by PGijsbers** [comment]
You're right, I didn't look closely enough at the rest of the script and just put on a band-aid. I opened an issue [in their repo](https://github.com/sberbank-ai-lab/LightAutoML/issues/57) to clarify. I'll hold off merging this for now.

**Comment by sebhrusen** [comment]
```
for TabularUtilizedAutoML (as it is a composition of TabularAutoMLs) it is a little bit harder - automl.outer_pipes[0].ml_algos[0].models[0].reader.class_mapping, as we go to the inner automl model and take its Reader property.
```
it looks like lightautoml integration needs a fix

**Comment by PGijsbers** [comment]
Agreed. The `probability_labels` should be passed and the integers converted to labels.

It still leaves the original question of why `y_test` is available to the frameworks to begin with, though that is outside the scope of this PR.

**Comment by PGijsbers** [comment]
Improved, predictions are now always converted back to their original domain and `probabilities_labels` is given. We might still want to wait for a response on https://github.com/sberbank-ai-lab/LightAutoML/issues/49 before merging.

File: frameworks/lightautoml/exec.py
Comment (sebhrusen) [inline frameworks/lightautoml/exec.py:61]: If I understand that `col` is always an integer, then the following is the only way to guarantee that `probabilities_labels` will be sorted correctly:
```suggestion
            probabilities_labels = [class_map[k] for k in sorted(class_map)]
```

Comment (PGijsbers) [reply frameworks/lightautoml/exec.py:61]: `class_map` is a dictionary from class to column, e.g. `{"b": 0, "a": 1}` sorted in order of columns.
Should `probabilities_labels` not match the order of the columns of `probabilities`? `sorted(class_map)` will just put the labels in alphabetical order. If you mean that you don't want to rely on the order of `class_map` then I suppose you'd want `[column_to_class[col] for col in sorted(column_to_class)]`?

Comment (sebhrusen) [reply frameworks/lightautoml/exec.py:61]: > class_map is a dictionary from class to column, e.g. {"b": 0, "a": 1} sorted in order of columns.

is it guaranteed by contract? as hidden implementation detail?

> [column_to_class[col] for col in sorted(column_to_class)]?

yes, that's what I meant, sorry.

Comment (PGijsbers) [reply frameworks/lightautoml/exec.py:61]: I can't find `class_mapping` in their documentation at all so I will assume it's an implementation detail and explicitly ensure the order.

Comment (sebhrusen) [reply frameworks/lightautoml/exec.py:61]: thanks

Comment (PGijsbers) [reply frameworks/lightautoml/exec.py:61]:
```suggestion
            probabilities_labels = [column_to_class[col] for col in sorted(column_to_class)]
```

**Review by sebhrusen — APPROVED** [review]

**Pull request merged**
